src/components/MovieListPage.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import Search from './Search';
+import MovieListItem from './MovieListItem';
+class MovieListPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filterText: '',
+    };
+  }
+
+  handleFilterTextChange = (value) => {
+    this.setState({ filterText: value });
+  };
+
+  handleSelectMovie = (imdbID) => {
+    this.props.onSelectMovie(imdbID);
+  };
+
+  propertiesContainFilterText(propertiesArray) {
+    const lowercaseFilterText = this.state.filterText.toLowerCase();
+    return propertiesArray.some((property) =>
+      property.toLowerCase().includes(lowercaseFilterText)
+    );
+  }
+
+  render() {
+    const movieList = [];
+    this.props.movies.forEach((movie) => {
+      if (this.propertiesContainFilterText([movie.Title, movie.Plot])) {
+        movieList.push(
+          <MovieListItem
+            key={movie.imdbID}
+            movie={movie}
+            onSelectMovie={this.handleSelectMovie}
+          />
+        );
+      }
+    });
```

Comment:
I would replace it with `map` function, it would be cleaner and more declarative.

Reply:
Fixed